The prompt to enter the site admin username and password is working but when I commented out the lines as requested it can no longer access the client secret (it isn't in the `env.sh` file) so I get this error message:
```
getting site admin token
Enter username: site_admin@test.ca
Enter password: 
Traceback (most recent call last):
  File "/Users/marionshadbolt/Documents/GitHub/CanDIGv2/site_admin_token.py", line 29, in get_site_admin_token
    client_secret=ENV["CANDIG_CLIENT_SECRET"],
                  ~~~^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'CANDIG_CLIENT_SECRET'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/marionshadbolt/Documents/GitHub/CanDIGv2/site_admin_token.py", line 58, in <module>
    result = get_site_admin_token()
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/marionshadbolt/Documents/GitHub/CanDIGv2/site_admin_token.py", line 55, in get_site_admin_token
    raise authx.auth.CandigAuthError(f"Error obtaining response from keycloak server: {e}")
authx.auth.CandigAuthError: Error obtaining response from keycloak server: 'CANDIG_CLIENT_SECRET'
```
there is no `tmp/keycloak/client-secret` file in my tmp directory

--- a/Makefile
+++ b/Makefile
@@ -567,22 +567,24 @@ start-all:
 
 #>>>
 # rebuild the entire stack without touching the data containers, defined in .env
+### $(MAKE) clean-all CANDIG_MODULES="$(CANDIG_MODULES)"
 #<<<
 
 .PHONY: rebuild-keep-data
 rebuild-keep-data:
-	# Remove the module from the .env
-	$(eval CANDIG_MODULES := $(filter-out $(CANDIG_DATA_MODULES),$(CANDIG_MODULES)))
-	# Clean everything
-	$(MAKE) clean-all CANDIG_MODULES="$(CANDIG_MODULES)"
+	# Remove the data modules from CANDIG_MODULES
+	$(eval REBUILD_CANDIG_MODULES := $(filter-out $(CANDIG_DATA_MODULES),$(CANDIG_MODULES)))
+	# Clean only the remaining modules
+	$(foreach MODULE, $(REBUILD_CANDIG_MODULES), $(MAKE) clean-$(MODULE);)
+	# Prune unused Docker resources
 	docker system prune -af
 	# Start build-all
 	./pre-build-check.sh $(ARGS)
 	$(MAKE) init-docker
-	# Rebuild everything
-	$(foreach MODULE, $(CANDIG_MODULES), $(MAKE) build-$(MODULE); $(MAKE) compose-$(MODULE);)
+	# Rebuild deleted modules
+	$(foreach MODULE, $(REBUILD_CANDIG_MODULES), $(MAKE) build-$(MODULE); $(MAKE) compose-$(MODULE);)
+	# Run post-build tasks
 	./post_build.sh
-
 
 # wrapper for make_backup.sh to make sure we're running it from the right directory
 backup-vault:

